torch_geometric/loader/utils.py

```diff
@@ -199,10 +207,43 @@ def filter_custom_store(
 # Input Utilities #############################################################
 
 
+@dataclass
+class SamplingInputNodes:
+    input_nodes: Dict[Optional[str], Sequence]
+
+    @property
+    def node_types(self) -> Tuple[Optional[str]]:
+        return tuple(self.input_nodes.keys())
+
+    @property
+    def node_type(self) -> Optional[str]:
+        node_types = self.node_types
+        if len(node_types) != 1:
+            raise ValueError(
+                "Requested 'node_type' but there are multiple types in nodes")
+        return node_types[0]
+
+    def nodes(self, node_type: NodeType) -> OptTensor:
+        return self.input_nodes.get(node_type, None)
+
+    @property
+    def as_list(self) -> Tuple[Tuple[str, int]]:
+        return tuple([(node_type, int(i))
+                      for node_type, tensor in self.input_nodes.items()
+                      for i in tensor])
```

Comment:
I guess we need this (and the below function) so that we can pass the right data to the PyTorch DataLoader constructor; perhaps we can leave a note here explaining that?